lib/workers/repository/update/pr/index.ts

```diff
@@ -249,7 +287,7 @@ export async function ensurePr(
     }
   }
 
-  const prBody = await getPrBody(config);
+  let prBody = await getPrBody(config);
```

Comment:
prBody is being called always, i went for the Lazy approach where we do the calculations only in case we need to update/create pr.

putting it in `getPrBody` will require also existingPr to get the bodyStruct or should i do it differently and change all code?... not to mention the extra calls always instead of lazy approach

Reply:
and if im inside getPrBody, how would i know if its an EXISTING PR or not, 
think about backward compatibility, i dont want to find that i dont have debug data, and then write there Creation version = This version and then its wrong, maybe the PR is old before the feature...

Reply:
because one of the limitations is what will we do with old PRs that are open didnt have it yet

Reply:
ofcourse i thought, ok lets just add it as if it exists in this case
i added 
```
if(existingPr) {
    if(!existingPr.bodyStruct?.debugData){
        existingPr.bodyStruct?.debugData = {updatedByRenovateVersion: pkg.version};
    }
  }
```

but apparently PrBodyStruct is optional and sometimes it doesnt exist which makes things more complicated to me

Reply:
or i can just send a flag `existingPr` true or false, and it would solve all my problems

Reply:
nevermind, so what i see now is that in `getPrBody` i will handle Updates only with existingPr/non existing Pr
and i will render Creation only on PR creation